platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -167,33 +175,57 @@ public void destroy() throws ApplicationServiceException {
     }
   }
 
+  @Override
+  public void installProfile(String profile, boolean noRefresh) {
+    try {
+      EnumSet<FeaturesService.Option> options = EnumSet.noneOf(FeaturesService.Option.class);
+      if (noRefresh) {
+        options.add(FeaturesService.Option.NoAutoRefreshBundles);
+      }
+      featuresService.installFeature(profile, options);
+    } catch (Exception e) {
```

Comment:
❓Should we replace all those new `catch (Exception e )` with `catch (RuntimeException e)` in case new checked exceptions get thrown in the `try` block at some point and need to be handled differently?

Reply:
That's an option for methods that don't use the features service. Unfortunately the Karaf features service throws `Exception`. We might want to keep it so we have consistent behavior across the MBean until otherwise necessary. If we only log `RuntimeException`s then we're not reporting all the failure information.